test/mock/grpc/mock_management_command_service.go

```diff
@@ -136,8 +136,34 @@ func (cs *CommandService) UpdateDataPlaneHealth(
 }
 
 func (cs *CommandService) Subscribe(in v1.CommandService_SubscribeServer) error {
-	go func() {
-		for {
+	ctx := in.Context()
+
+	go cs.listenForDataPlaneResponses(ctx, in)
+
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		default:
+			slog.Info("Starting Subscribe")
+			request := <-cs.requestChan
+
+			slog.Debug("Subscribe", "request", request)
+
+			err := in.Send(request)
+			if err != nil {
+				slog.Error("Failed to send management request", "error", err)
+			}
+		}
+	}
+}
+
+func (cs *CommandService) listenForDataPlaneResponses(ctx context.Context, in v1.CommandService_SubscribeServer) {
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		default:
 			dataPlaneResponse, err := in.Recv()
```

Comment:
I realized something kinda fundamental, have actually defined the receive and send streams the wrong way around in our Protos?
```
// A decoupled communication mechanism between the data plane and management plane.
// buf:lint:ignore RPC_RESPONSE_STANDARD_NAME
// buf:lint:ignore RPC_REQUEST_STANDARD_NAME
rpc Subscribe(stream DataPlaneResponse) returns (stream ManagementPlaneRequest) {}
```
We should be receiving ManagementPlaneRequests and sending DataPlaneResponses from the subscription (`v1.CommandService_SubscribeServer` in this case) but instead in our code it's vice versa.

Reply:
As @aphralG correctly pointed out, this is the mock management plane, so it's Subscribe is naturally gonna be the reverse of the Agent's:
```go
type CommandService_SubscribeServer interface {
	Send(*ManagementPlaneRequest) error
	Recv() (*DataPlaneResponse, error)
	grpc.ServerStream
}
```
Resolving this comment and grabbing more coffee! ☕